### PR TITLE
fix(@deaktop/wallet): Fix transfer function amount

### LINKED ISSUE #8813: Fix transfer function amount
In case of multi tx, the amount is the total amount and not the one that should be split in multiple tx

Also this is a good chance to refactor this complex function

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
fixes #8813

### What does the PR do

1. Simplifies the breaks down the transfer function and
2. For multi tx the corresponding amount to each chain is sent with the request

### Affected areas

Wallet

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### CODE REVIEW

File: src/app_service/service/transaction/service.nim [hunk 1]
```diff
@@ -230,105 +230,150 @@ QtObject:
...
+      toAddress,
+      $route.gasAmount,
+      gasFees,
+      route.gasFees.eip1559Enabled,
+      $route.gasFees.maxPriorityFeePerGas,
+      $route.gasFees.maxFeePerGasM
+    )
+    txData.data = data
+
+    var path = TransactionBridgeDto(bridgeName: "Simple", chainID: route.fromNetwork.chainId)
```
Comment (jrainville) [inline src/app_service/service/transaction/service.nim:251]: Those names should be put in constants IMO

File: src/app_service/service/transaction/service.nim [hunk 2]
```diff
@@ -230,105 +230,150 @@ QtObject:
...
+        var gasFees: string = ""
+
+        if( not route.gasFees.eip1559Enabled):
+          gasFees = $route.gasFees.gasPrice
+
+        if route.approvalRequired:
+          paths.add(self.createApprovalPath(route, from_addr, toAddress, gasFees))
+
+        txData = ens_utils.buildTransaction(parseAddress(from_addr), route.amountIn,
+  $route.gasAmount, gasFees, route.gasFees.eip1559Enabled, $route.gasFees.maxPriorityFeePerGas, $route.gasFees.maxFeePerGasM)
```
Comment (jrainville) [inline src/app_service/service/transaction/service.nim:306]: Something weird happened with indentation here.

**Review by jrainville — COMMENTED** [review]
Looks good, two small comments

**Review by alaibe — APPROVED** [review]

File: src/app_service/service/transaction/service.nim (hunk 1 above)
Comment (Khushboo-dev-cpp) [reply src/app_service/service/transaction/service.nim]: done!

File: src/app_service/service/transaction/service.nim (hunk 2 above)
Comment (Khushboo-dev-cpp) [reply src/app_service/service/transaction/service.nim]: fixed :)

**Comment by Khushboo-dev-cpp** [comment]
@jrainville ive fixed you comments check when you can :)

**Review by jrainville — APPROVED** [review]

**Pull request merged**